sdk/keyvault/keyvault-secrets/package.json

```diff
@@ -114,35 +107,59 @@
     "tslib": "^2.2.0"
   },
   "devDependencies": {
-    "@azure-tools/test-credential": "^1.0.0",
-    "@azure-tools/test-recorder": "^3.0.0",
-    "@azure-tools/test-utils": "^1.0.1",
+    "@azure-tools/test-credential": "^2.0.0",
+    "@azure-tools/test-recorder": "^4.1.0",
+    "@azure-tools/test-utils-vitest": "^1.0.0",
     "@azure/dev-tool": "^1.0.0",
     "@azure/eslint-plugin-azure-sdk": "^3.0.0",
     "@azure/identity": "^4.0.1",
     "@microsoft/api-extractor": "^7.31.1",
-    "@types/mocha": "^10.0.0",
     "@types/node": "^18.0.0",
-    "@types/sinon": "^17.0.0",
+    "@vitest/coverage-istanbul": "^2.1.1",
     "cross-env": "^7.0.2",
     "dotenv": "^16.0.0",
     "eslint": "^9.9.0",
-    "karma": "^6.2.0",
-    "karma-chrome-launcher": "^3.0.0",
-    "karma-coverage": "^2.0.0",
-    "karma-env-preprocessor": "^0.1.1",
-    "karma-firefox-launcher": "^1.1.0",
-    "karma-junit-reporter": "^2.0.1",
-    "karma-mocha": "^2.0.1",
-    "karma-mocha-reporter": "^2.2.5",
-    "karma-sourcemap-loader": "^0.3.8",
-    "mocha": "^10.0.0",
-    "nyc": "^17.0.0",
-    "puppeteer": "^23.0.2",
+    "playwright": "^1.47.2",
     "rimraf": "^5.0.5",
-    "sinon": "^17.0.0",
-    "source-map-support": "^0.5.9",
-    "ts-node": "^10.0.0",
-    "typescript": "~5.6.2"
+    "tshy": "^3.0.2",
```

Comment:
Should we use v2 instead? see context in https://github.com/Azure/azure-sdk-for-js/pull/30740#discussion_r1715903828

Reply:
Yes, and actually dev-tool run build-package uses vendored tshy so I just removed tshy from here